doc/workshops/scientific-computing.rst

```diff
@@ -2,4 +2,753 @@
+
+To appreciate in context the real difference in the averages, let's create some bar charts. We are now going to use Seaborn's barplot class. On the x-axis we put again the qualitative variable (q997) and on the y-axis the quantitative variable (cognitive_score). The data parameter takes as argument the entire dataset.
+
+.. code-block:: python
+
+   fig_dims = (12, 8)
+   fig, ax = plt.subplots(figsize=fig_dims)
+   sns.barplot(x=ca_test.q997, y=ca_test.cognitive_score, ax=ax, data=ca_test)
+   sns.set_palette(palette)
+   plt.xlabel("q997 = Are you a cat person or a dog person?")
+   plt.ylabel("cognitive score")
+
+
+.. image:: images/workshop-2/pet-preferences-cognitive-score-bar-graph.png
+   :alt: Pet preferences and cognitive score mean representation (box graph)
+
+If the box plots indicated that the means did not show significant differences (possibly due to chance), the bar plot confirms that these differences are, in proportion, quite modest.
+
+
+
+==========
+ANOVA test
+==========
+
+To make sure that the differences between the means are statistically significant, we will perform an ANOVA (analysis of variance) because we are going to compare the means of more than two groups. 
+
+How does ANOVA work? This analysis tells us whether three or more means are equal. If so, this would support the null hypothesis.
+
+.. note::
+
+   The *null hypothesis* is an alternative prediction to the one we have stated in our hypothesis. The null hypothesis simply states that the observed differences in means are due to random variations that occur when samples are collected.
+
+An ANOVA produces an *F-ratio* (also called F-statistic) that compares the amount of systematic variation in the data (variation that can be explained by the model or hypothesis) with the amount of unsystematic variance (variation that cannot be explained by our hypothesis or model). This means that F is the ratio of the model to its error. ANOVA also produces a *p-value* that indicates the probability that the variation can be attributed to the null hypothesis. The smaller the p-value, the less likely it is that the observed variation is due to chance.
+
+To perform an ANOVA, let's start by pivoting the dataset with the cognitive ability test results. The pet preferences values will now be the column names, and each row will represent a person.
+
+.. code-block:: python
+
+   dog_or_cat_pivot = ca_test.pivot(columns="q997", values="cognitive_score")
+
+You will see that one of the columns includes the answers to the test made by people who did not specify their preferences towards pets. Let's delete that column.
+
+.. code-block:: python
+
+   dog_or_cat_pivot.drop(dog_or_cat_pivot.columns[0], axis=1, inplace=True)
+
+.. image:: images/workshop-2/pets-preferences-pivot-table
```

Comment:
I think the extension of the file is missing here (`.png`)

Reply:
Yeap the image is not working properly